modules/users/server/controllers/users/users.profile.server.controller.js

```diff
@@ -109,24 +135,54 @@ exports.changeProfilePicture = function (req, res) {
   function deleteOldImage() {
     return new Promise(function (resolve, reject) {
       if (existingImageUrl !== User.schema.path('profileImageURL').defaultValue) {
-        fs.unlink(existingImageUrl, function (unlinkError) {
-          if (unlinkError) {
+        if (config.uploads.storage === 's3' && config.aws.s3) {
+          try {
+            var { region, bucket, key } = amazonS3URI(existingImageUrl);
```

Comment:
This could be troublesome, if switching from "local" to "s3" storage. The first time a user updates their profile picture after switching the storage from one to another, this will create inconsistency with the data/files. This is probably more of an edge case, or migration consideration. Perhaps not a big deal?

Reply:
I thought about that as well. Normally you have to configure your app once!
An edge case for me.